core/web/resolver/testdata/config-multi-chain-effective.toml

```diff
@@ -432,8 +432,8 @@ SamplingInterval = '1s'
 [EVM.NodePool]
 PollFailureThreshold = 5
 PollInterval = '10s'
-SelectionMode = 'HighestHead'
-SyncThreshold = 10
+SelectionMode = 'TotalDifficulty'
+SyncThreshold = 250
```

Comment:
Why has this changed?

Reply:
IIUC because this represents a difference in difficulty when in difficulty mode, and 250 = 10 block * max of 25 per

Perhaps we should rethink whether using difficulty for `SyncThreshold` makes sense :thinking: 
The trouble is that block height math is a little odd too - e.g. we only care about block height difference from the height of the greatest total difficulty, not necesssarily the highest block.

Reply:
If we're talking about this specific file, one of the tests that constructs a default config and compares it against an expected config was failing and had to be updated with the revised Polygon values:
https://github.com/smartcontractkit/chainlink/actions/runs/6036836393/job/16379921901

Talking about `SyncThreshold` in general I agree with Jordan. The values are currently set this way just to retain the previous behavior of 5/10 blocks as close as possible.

Reply:
For now, this logic seems fine, although agree that we should rethink how to do this for TotalDifficulty.

I'm also not confident our existing default values for SyncThreshold were set correctly for all chains. I see most ones have 5 or 10 set. Ideally faster chains should have a larger limit here. Maybe something for us to tackle later on, to research and set right values for default configs like this one.

Reply:
Actually looking again, for TotalDifficulty, we use SyncThreshold to find difference between TTD of best node, vs current node. So if a node is waay behind on TTD from best node, then makes sense to mark it as OutOfSync. I don't think the logic is odd or incorrect.

Reply:
The happy path works fine, but if some signers are down, and difficulty accumulates more slowly over a period of time, then we end up incidentally waiting longer (by wall time and block height) to mark nodes out-of-sync. The opposite is true if the signer set was permanently expanded - until reconfigured, nodes would be marked out-of-sync more quickly.

Reply:
We could explore different designs, if we determine that current logic is flawed, and we see issues in the wild.
I have a few ideas, that we could entertain:
1) Set the SyncThreshold as a percentile of the TotalDifficulty of best node minus finality node. So if best node is 100 difficulty ahead of finality node, then we say any other node with less than 60 is out of sync.
2) Do away with this concept of out of sync, if behind the best head. Instead, at all times track who the best node is, and rank others based on their distance from best head. For any RPC calls, just choose the best node at that particular time.

Reply:
> For any RPC calls, just choose the best node at that particular time.

I don't think this part is viable, because the it can thrash the node's view around, and triggers expensive/disruptive ops like re-subscribing to websockets each time.

Reply:
I think that doing simple block height math from the greatest total difficulty node would better align things in the short term. Longer term, I wonder if being able to express this as a wall clock time instead might be desirable from a UX perspective.